src/main/java/com/google/sps/dao/AssignmentDao.java

```diff
@@ -89,4 +94,112 @@ public ValidationResponse createAssignment(Assignment assignment) {
         }
         return assignments;
     }
+
+    @Override
+    @SneakyThrows
+    public ValidationResponse updateAssignmentForStudent(Assignment newAssignment) {
+
+        ValidationResponse.Builder validationResponseBuilder = null;
+
+        Assignment oldAssignment = getAssignmentById(newAssignment.getId());
+        newAssignment.setId(oldAssignment.getId());
+        newAssignment.setUpdated(System.currentTimeMillis());
+
+        if (ValidationUtils.isEmptyOrWhiteSpace(newAssignment.getSolution())) {
+            validationResponseBuilder = ValidationResponse.builder();
+            validationResponseBuilder.status(ValidationErrors.STATUS_NOT_OK);
+            validationResponseBuilder.message(ValidationErrors.MESSAGE_NOT_OK);
+
+            validationResponseBuilder.error(
+                    ValidationErrors.isEmptyOrWhiteSpace(Assignment.Keys.SOLUTION)
+            );
+        }
+
+        //do not set scored_marks attribute
+        //solution attribute is set from the request
+        newAssignment.setCreated(oldAssignment.getCreated());
+        newAssignment.setAssignmentId(oldAssignment.getAssignmentId());
+        newAssignment.setClassroomId(oldAssignment.getClassroomId());
+        newAssignment.setEducatorId(oldAssignment.getEducatorId());
+        newAssignment.setDeadline(oldAssignment.getDeadline());
+        newAssignment.setTotal_marks(oldAssignment.getTotal_marks());
+        newAssignment.setName(oldAssignment.getName());
+        newAssignment.setDescription(oldAssignment.getDescription());
+
+        //Set attribute submitted to true
+        newAssignment.setSubmitted(true);
+
+        ApiFuture<WriteResult> writeResult = db.collection(Assignment.Keys.COLLECTION)
+                .document(newAssignment.getId()).set(newAssignment);
+
+        log.info("Update assignment called: " + writeResult.get().toString());
+
+        return ValidationResponse.builder()
+                .status(ValidationErrors.STATUS_OK)
+                .message(ValidationErrors.MESSAGE_OK)
+                .build();
+
+    }
+
+    @Override
+    @SneakyThrows
+    public ValidationResponse updateAssignmentForEducator(Assignment newAssignment) {
+
+        ValidationResponse.Builder validationResponseBuilder = null;
+
+        Assignment oldAssignment = getAssignmentById(newAssignment.getId());
+        newAssignment.setId(oldAssignment.getId());
+        newAssignment.setUpdated(System.currentTimeMillis());
+
+        if(newAssignment.getScored_marks() < 0 || newAssignment.getScored_marks() > oldAssignment.getTotal_marks()){
+            validationResponseBuilder = ValidationResponse.builder();
+            validationResponseBuilder.status(ValidationErrors.STATUS_NOT_OK);
+            validationResponseBuilder.message(ValidationErrors.MESSAGE_NOT_OK);
```

Comment:
We can add an error message for the error as well. Something like - Score cannot be more than total marks.

Reply:
done